pipeline.go

```diff
@@ -50,26 +50,38 @@ func (c *Pipeline) Discard() error {
 
 // Exec always returns list of commands and error of the first failed
 // command if any.
-func (c *Pipeline) Exec() ([]Cmder, error) {
+func (c *Pipeline) Exec() (cmds []Cmder, retErr error) {
```

Comment:
I prefer your implmentation where pipeline is using MaxRetries too. Can you see any edge case where this would not be desirable?